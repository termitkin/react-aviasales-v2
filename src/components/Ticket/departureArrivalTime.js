/*
INPUT: date string, duration int
OUTPUT: "10:36 – 18:00"
*/

export const departureArrivalTime = (dateFromProps, durat) => {
  const date = new Date(dateFromProps);
  const datePlusDuration = date.getTime() + durat * 60000;

  let departureTimeHours =
    date.getHours() > 9 ? date.getHours() : `0${date.getHours()}`;
  let departureTimeMinutes =
    date.getMinutes() > 9 ? date.getMinutes() : `0${date.getMinutes()}`;
  const departureTime = `${departureTimeHours}:${departureTimeMinutes}`;

  let ArrivalTimeHours =
    new Date(datePlusDuration).getHours() > 9
      ? new Date(datePlusDuration).getHours()
      : `0${new Date(datePlusDuration).getHours()}`;
  let ArrivalTimeMinutes =
    new Date(datePlusDuration).getMinutes() > 9
      ? new Date(datePlusDuration).getMinutes()
      : `0${new Date(datePlusDuration).getMinutes()}`;
  const ArrivalTime = `${ArrivalTimeHours}:${ArrivalTimeMinutes}`;

  const result = `${departureTime} – ${ArrivalTime}`;

  return result;
};
