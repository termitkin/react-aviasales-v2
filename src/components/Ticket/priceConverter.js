/*
INPUT: int price - 83733
OUTPUT: "83 733 P"
*/

export const priceConverter = (price, currency) => {
  if (currency === "usd") {
    price = price / 66;
    price = Number(price).toFixed(2);
    price = price.toString().replace(/\B(?=(\d{3})+(?!\d))/g, " ") + " $";
  } else if (currency === "eur") {
    price = price / 75;
    price = Number(price).toFixed(2);
    price = price.toString().replace(/\B(?=(\d{3})+(?!\d))/g, " ") + " €";
  } else {
    price = price.toString().replace(/\B(?=(\d{3})+(?!\d))/g, " ") + " ₽";
  }

  return price;
};
