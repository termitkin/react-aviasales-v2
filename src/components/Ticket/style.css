.no-tickets-message {
  color: #333;
  text-align: center;
}

.no-tickets-message_show_true {
  display: block;
}

.no-tickets-message_show_false {
  display: none;
}