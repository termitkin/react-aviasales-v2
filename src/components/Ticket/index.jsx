import React from "react";
import PropTypes from "prop-types";
import styled from "@emotion/styled/macro";
import { device } from "../mediaQueries";
import { duration } from "./duration";
import { departureArrivalTime } from "./departureArrivalTime";
import { stopsText } from "./stopsText";
import { priceConverter } from "./priceConverter";
import "./style.css";

const StyledTicket = styled.div`
  width: 100%;

  box-shadow: 0px 2px 8px rgba(0, 0, 0, 0.1);
  border-radius: 5px;
  padding: 5px;
  margin-top: 20px;
  box-sizing: border-box;
  background-color: #fff;
  border-bottom: 3px solid transparent;
  transition: border-bottom 0.25s;

  &:hover,
  &:focus {
    cursor: pointer;
    border-bottom: 3px solid #2196f3;
  }

  @media ${device.desktop} {
    width: 100%;

    box-shadow: 0px 2px 8px rgba(0, 0, 0, 0.1);
    border-radius: 5px;
    padding: 20px;
    margin-top: 20px;
    box-sizing: border-box;
    background-color: #fff;
  }
`;

const StyledShadow = styled.div`
  position: relative;
  z-index: 0;
  box-shadow: 0px 1px 4px rgba(91, 137, 164, 0.25);
  border-radius: 5px;
  transition: box-shadow 0.25s;

  &:hover,
  &:focus {
    outline: none;
    box-shadow: 0px 5px 25px rgba(91, 137, 164, 0.1);
  }

  &:hover ${StyledTicket}, &:focus ${StyledTicket} {
    border-bottom: 3px solid #2196f3;
  }

  &:after {
    content: "";
    width: 100%;
    height: 46px;
    position: absolute;
    box-shadow: 0px 10px 50px rgba(91, 137, 164, 0.25);
    bottom: 0;
    left: 50%;
    transform: translate(-50%, 0);
    z-index: -1;
    opacity: 0;
    transition: opacity 0.25s;
    border-radius: 5px;
  }

  &:hover:after,
  &:focus:after {
    content: "";
    width: 100%;
    height: 46px;
    position: absolute;
    box-shadow: 0px 10px 50px rgba(91, 137, 164, 0.25);
    bottom: 0;
    left: 50%;
    transform: translate(-50%, 0);
    z-index: -1;
    opacity: 1;
  }
`;

const StyledPriceAndLogo = styled.div`
  width: 100%;
  height: 36px;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const StyledLogoWrap = styled.picture`
  width: 140px;
  height: 36px;
  display: flex;
  justify-content: flex-end;
`;

const StyledLogo = styled.img`
  width: 99px;
  height: 36px;
  margin-right: 5px;

  @media ${device.desktop} {
    margin-right: 0px;
  }
`;

const StyledPrice = styled.div`
  margin-left: 5px;
  font-family: Open Sans;
  font-style: normal;
  font-weight: 600;
  font-size: 24px;
  line-height: 24px;
  color: #2196f3;

  @media ${device.desktop} {
    margin-left: 0;
  }
`;

const StyledContentWrap = styled.div`
  width: 33.3%;
  display: flex;
  align-items: center;
  justify-content: center;
  font-family: Open Sans;
  font-style: normal;
  font-weight: 600;

  @media ${device.desktop} {
    width: 140px;
    justify-content: flex-start;
  }
`;

const StyledMeta = styled(StyledContentWrap)`
  font-size: 12px;
  height: 18px;
  color: #a0b0b9;
`;

const StyledContent = styled(StyledContentWrap)`
  font-size: 13px;
  height: 21px;
  color: #4a4a4a;

  @media ${device.desktop} {
    font-size: 14px;
    height: 21px;
    color: #4a4a4a;
  }
`;

const StyledSegment = styled.div`
  width: 100%;
  height: 39px;
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  align-content: space-between;
  margin-top: 20px;

  &:last-child {
    margin-top: 10px;
  }
`;

const Ticket = props => {
  const ticket = props.ticket.map(el => {
    return (
      <StyledShadow
        key={
          el.carrier +
          el.price +
          el.segments[0].origin +
          el.segments[0].duration +
          el.segments[1].duration
        }
        tabIndex="0"
      >
        <StyledTicket>
          <StyledPriceAndLogo>
            <StyledPrice>
              {priceConverter(
                el.price,
                props.currentCurrency,
                props.currencyRates
              )}
            </StyledPrice>
            <StyledLogoWrap>
              <source
                srcSet={`https://pics.avs.io/99/36/${el.carrier}@2x.png`}
                type="image/png"
                media="screen and (-webkit-min-device-pixel-ratio: 1.25), screen and ( min--moz-device-pixel-ratio: 1.25), screen and (-o-min-device-pixel-ratio: 1.25/1), screen and (min-device-pixel-ratio: 1.25), screen and ( min-resolution: 200dpi), screen and ( min-resolution: 1.25dppx)"
              ></source>
              <StyledLogo
                src={`https://pics.avs.io/99/36/${el.carrier}.png`}
                alt="Логотип авиакомпании"
              />
            </StyledLogoWrap>
          </StyledPriceAndLogo>

          <StyledSegment>
            <StyledMeta>{`${el.segments[0].origin} – ${el.segments[0].destination}`}</StyledMeta>
            <StyledContent>
              {departureArrivalTime(
                el.segments[0].date,
                el.segments[0].duration
              )}
            </StyledContent>
            <StyledMeta>В пути</StyledMeta>
            <StyledContent>{duration(el.segments[0].duration)}</StyledContent>
            <StyledMeta>{stopsText(el.segments[0].stops)}</StyledMeta>
            <StyledContent>
              {[...el.segments[0].stops].join(", ")}
            </StyledContent>
          </StyledSegment>
          <StyledSegment>
            <StyledMeta>{`${el.segments[1].origin} – ${el.segments[1].destination}`}</StyledMeta>
            <StyledContent>
              {departureArrivalTime(
                el.segments[1].date,
                el.segments[1].duration
              )}
            </StyledContent>
            <StyledMeta>В пути</StyledMeta>
            <StyledContent>{duration(el.segments[1].duration)}</StyledContent>
            <StyledMeta>{stopsText(el.segments[1].stops)}</StyledMeta>
            <StyledContent>
              {[...el.segments[1].stops].join(", ")}
            </StyledContent>
          </StyledSegment>
        </StyledTicket>
      </StyledShadow>
    );
  });

  return ticket;
};

export default Ticket;

Ticket.propTypes = {
  ticket: PropTypes.array.isRequired,
  currentCurrency: PropTypes.string.isRequired,
  currencyRates: PropTypes.object.isRequired
};
