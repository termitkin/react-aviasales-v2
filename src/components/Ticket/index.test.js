import React from "react";
import { create, act } from "react-test-renderer";
import Ticket from "./index";

describe("Ticket component", () => {
  let wrapper;

  afterAll(() => {
    wrapper.unmount();
  });

  it("Matches the snapshot", () => {
    act(() => {
      wrapper = create(
        <Ticket
          ticket={[
            {
              carrier: "TG",
              price: 15005,
              segments: [
                {
                  date: "2019-10-11T06:51:00.000Z",
                  destination: "HKT",
                  duration: 1557,
                  origin: "MOW",
                  stops: []
                },
                {
                  date: "2019-10-31T11:39:00.000Z",
                  destination: "HKT",
                  duration: 752,
                  origin: "MOW",
                  stops: ["BKK", "AUH", "HKG"]
                }
              ]
            }
          ]}
          currentCurrency={"rub"}
          currencyRates={{
            eur: 70.7557,
            usd: 64.9790614381
          }}
        />
      );
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it("Matches the snapshot 2", () => {
    act(() => {
      wrapper = create(
        <Ticket
          ticket={[
            {
              carrier: "TG",
              price: 25000,
              segments: [
                {
                  date: "2019-10-11T06:51:00.000Z",
                  destination: "HKT",
                  duration: 1557,
                  origin: "MOW",
                  stops: []
                },
                {
                  date: "2019-10-31T11:39:00.000Z",
                  destination: "HKT",
                  duration: 752,
                  origin: "MOW",
                  stops: ["BKK", "AUH", "HKG"]
                }
              ]
            }
          ]}
          currentCurrency={"eur"}
          currencyRates={{
            eur: 51,
            usd: 63
          }}
        />
      );
    });

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
