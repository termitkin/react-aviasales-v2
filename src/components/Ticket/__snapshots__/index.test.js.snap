// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticket component Matches the snapshot 1`] = `
.emotion-38 {
  position: relative;
  z-index: 0;
  box-shadow: 0px 1px 4px rgba(91,137,164,0.25);
  border-radius: 5px;
  -webkit-transition: box-shadow 0.25s;
  transition: box-shadow 0.25s;
}

.emotion-38:hover,
.emotion-38:focus {
  outline: none;
  box-shadow: 0px 5px 25px rgba(91,137,164,0.1);
}

.emotion-38:hover .emotion-37,
.emotion-38:focus .emotion-37 {
  border-bottom: 3px solid #2196f3;
}

.emotion-38:after {
  content: "";
  width: 100%;
  height: 46px;
  position: absolute;
  box-shadow: 0px 10px 50px rgba(91,137,164,0.25);
  bottom: 0;
  left: 50%;
  -webkit-transform: translate(-50%,0);
  -ms-transform: translate(-50%,0);
  transform: translate(-50%,0);
  z-index: -1;
  opacity: 0;
  -webkit-transition: opacity 0.25s;
  transition: opacity 0.25s;
  border-radius: 5px;
}

.emotion-38:hover:after,
.emotion-38:focus:after {
  content: "";
  width: 100%;
  height: 46px;
  position: absolute;
  box-shadow: 0px 10px 50px rgba(91,137,164,0.25);
  bottom: 0;
  left: 50%;
  -webkit-transform: translate(-50%,0);
  -ms-transform: translate(-50%,0);
  transform: translate(-50%,0);
  z-index: -1;
  opacity: 1;
}

.emotion-36 {
  width: 100%;
  box-shadow: 0px 2px 8px rgba(0,0,0,0.1);
  border-radius: 5px;
  padding: 5px;
  margin-top: 20px;
  box-sizing: border-box;
  background-color: #fff;
  border-bottom: 3px solid transparent;
  -webkit-transition: border-bottom 0.25s;
  transition: border-bottom 0.25s;
}

.emotion-36:hover,
.emotion-36:focus {
  cursor: pointer;
  border-bottom: 3px solid #2196f3;
}

@media (min-width:960px) {
  .emotion-36 {
    width: 100%;
    box-shadow: 0px 2px 8px rgba(0,0,0,0.1);
    border-radius: 5px;
    padding: 20px;
    margin-top: 20px;
    box-sizing: border-box;
    background-color: #fff;
  }
}

.emotion-6 {
  width: 100%;
  height: 36px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  margin-left: 5px;
  font-family: Open Sans;
  font-style: normal;
  font-weight: 600;
  font-size: 24px;
  line-height: 24px;
  color: #2196f3;
}

@media (min-width:960px) {
  .emotion-0 {
    margin-left: 0;
  }
}

.emotion-4 {
  width: 140px;
  height: 36px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.emotion-2 {
  width: 99px;
  height: 36px;
  margin-right: 5px;
}

@media (min-width:960px) {
  .emotion-2 {
    margin-right: 0px;
  }
}

.emotion-20 {
  width: 100%;
  height: 39px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: space-between;
  -ms-flex-line-pack: space-between;
  align-content: space-between;
  margin-top: 20px;
}

.emotion-20:last-child {
  margin-top: 10px;
}

.emotion-8 {
  width: 33.3%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Open Sans;
  font-style: normal;
  font-weight: 600;
  font-size: 12px;
  height: 18px;
  color: #a0b0b9;
}

@media (min-width:960px) {
  .emotion-8 {
    width: 140px;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

.emotion-10 {
  width: 33.3%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Open Sans;
  font-style: normal;
  font-weight: 600;
  font-size: 13px;
  height: 21px;
  color: #4a4a4a;
}

@media (min-width:960px) {
  .emotion-10 {
    width: 140px;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media (min-width:960px) {
  .emotion-10 {
    font-size: 14px;
    height: 21px;
    color: #4a4a4a;
  }
}

<div
  className="emotion-38 emotion-39"
  tabIndex="0"
>
  <div
    className="emotion-36 emotion-37"
  >
    <div
      className="emotion-6 emotion-7"
    >
      <div
        className="emotion-0 emotion-1"
      >
        15 005 ₽
      </div>
      <picture
        className="emotion-4 emotion-5"
      >
        <source
          media="screen and (-webkit-min-device-pixel-ratio: 1.25), screen and ( min--moz-device-pixel-ratio: 1.25), screen and (-o-min-device-pixel-ratio: 1.25/1), screen and (min-device-pixel-ratio: 1.25), screen and ( min-resolution:
 200dpi), screen and ( min-resolution: 1.25dppx)"
          srcSet="https://pics.avs.io/99/36/TG@2x.png"
          type="image/png"
        />
        <img
          alt="Логотип авиакомпании"
          className="emotion-2 emotion-3"
          src="https://pics.avs.io/99/36/TG.png"
        />
      </picture>
    </div>
    <div
      className="emotion-20 emotion-21"
    >
      <div
        className="emotion-8 emotion-9"
      >
        MOW – HKT
      </div>
      <div
        className="emotion-10 emotion-11"
      >
        09:51 – 11:48
      </div>
      <div
        className="emotion-8 emotion-9"
      >
        В пути
      </div>
      <div
        className="emotion-10 emotion-11"
      >
        25ч 57м
      </div>
      <div
        className="emotion-8 emotion-9"
      >
        0 пересадок
      </div>
      <div
        className="emotion-10 emotion-11"
      >
        
      </div>
    </div>
    <div
      className="emotion-20 emotion-21"
    >
      <div
        className="emotion-8 emotion-9"
      >
        MOW – HKT
      </div>
      <div
        className="emotion-10 emotion-11"
      >
        14:39 – 03:11
      </div>
      <div
        className="emotion-8 emotion-9"
      >
        В пути
      </div>
      <div
        className="emotion-10 emotion-11"
      >
        12ч 32м
      </div>
      <div
        className="emotion-8 emotion-9"
      >
        3 пересадки
      </div>
      <div
        className="emotion-10 emotion-11"
      >
        BKK, AUH, HKG
      </div>
    </div>
  </div>
</div>
`;

exports[`Ticket component Matches the snapshot 2 1`] = `
.emotion-38 {
  position: relative;
  z-index: 0;
  box-shadow: 0px 1px 4px rgba(91,137,164,0.25);
  border-radius: 5px;
  -webkit-transition: box-shadow 0.25s;
  transition: box-shadow 0.25s;
}

.emotion-38:hover,
.emotion-38:focus {
  outline: none;
  box-shadow: 0px 5px 25px rgba(91,137,164,0.1);
}

.emotion-38:hover .emotion-37,
.emotion-38:focus .emotion-37 {
  border-bottom: 3px solid #2196f3;
}

.emotion-38:after {
  content: "";
  width: 100%;
  height: 46px;
  position: absolute;
  box-shadow: 0px 10px 50px rgba(91,137,164,0.25);
  bottom: 0;
  left: 50%;
  -webkit-transform: translate(-50%,0);
  -ms-transform: translate(-50%,0);
  transform: translate(-50%,0);
  z-index: -1;
  opacity: 0;
  -webkit-transition: opacity 0.25s;
  transition: opacity 0.25s;
  border-radius: 5px;
}

.emotion-38:hover:after,
.emotion-38:focus:after {
  content: "";
  width: 100%;
  height: 46px;
  position: absolute;
  box-shadow: 0px 10px 50px rgba(91,137,164,0.25);
  bottom: 0;
  left: 50%;
  -webkit-transform: translate(-50%,0);
  -ms-transform: translate(-50%,0);
  transform: translate(-50%,0);
  z-index: -1;
  opacity: 1;
}

.emotion-36 {
  width: 100%;
  box-shadow: 0px 2px 8px rgba(0,0,0,0.1);
  border-radius: 5px;
  padding: 5px;
  margin-top: 20px;
  box-sizing: border-box;
  background-color: #fff;
  border-bottom: 3px solid transparent;
  -webkit-transition: border-bottom 0.25s;
  transition: border-bottom 0.25s;
}

.emotion-36:hover,
.emotion-36:focus {
  cursor: pointer;
  border-bottom: 3px solid #2196f3;
}

@media (min-width:960px) {
  .emotion-36 {
    width: 100%;
    box-shadow: 0px 2px 8px rgba(0,0,0,0.1);
    border-radius: 5px;
    padding: 20px;
    margin-top: 20px;
    box-sizing: border-box;
    background-color: #fff;
  }
}

.emotion-6 {
  width: 100%;
  height: 36px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  margin-left: 5px;
  font-family: Open Sans;
  font-style: normal;
  font-weight: 600;
  font-size: 24px;
  line-height: 24px;
  color: #2196f3;
}

@media (min-width:960px) {
  .emotion-0 {
    margin-left: 0;
  }
}

.emotion-4 {
  width: 140px;
  height: 36px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.emotion-2 {
  width: 99px;
  height: 36px;
  margin-right: 5px;
}

@media (min-width:960px) {
  .emotion-2 {
    margin-right: 0px;
  }
}

.emotion-20 {
  width: 100%;
  height: 39px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: space-between;
  -ms-flex-line-pack: space-between;
  align-content: space-between;
  margin-top: 20px;
}

.emotion-20:last-child {
  margin-top: 10px;
}

.emotion-8 {
  width: 33.3%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Open Sans;
  font-style: normal;
  font-weight: 600;
  font-size: 12px;
  height: 18px;
  color: #a0b0b9;
}

@media (min-width:960px) {
  .emotion-8 {
    width: 140px;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

.emotion-10 {
  width: 33.3%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Open Sans;
  font-style: normal;
  font-weight: 600;
  font-size: 13px;
  height: 21px;
  color: #4a4a4a;
}

@media (min-width:960px) {
  .emotion-10 {
    width: 140px;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media (min-width:960px) {
  .emotion-10 {
    font-size: 14px;
    height: 21px;
    color: #4a4a4a;
  }
}

<div
  className="emotion-38 emotion-39"
  tabIndex="0"
>
  <div
    className="emotion-36 emotion-37"
  >
    <div
      className="emotion-6 emotion-7"
    >
      <div
        className="emotion-0 emotion-1"
      >
        500.00 €
      </div>
      <picture
        className="emotion-4 emotion-5"
      >
        <source
          media="screen and (-webkit-min-device-pixel-ratio: 1.25), screen and ( min--moz-device-pixel-ratio: 1.25), screen and (-o-min-device-pixel-ratio: 1.25/1), screen and (min-device-pixel-ratio: 1.25), screen and ( min-resolution:
 200dpi), screen and ( min-resolution: 1.25dppx)"
          srcSet="https://pics.avs.io/99/36/TG@2x.png"
          type="image/png"
        />
        <img
          alt="Логотип авиакомпании"
          className="emotion-2 emotion-3"
          src="https://pics.avs.io/99/36/TG.png"
        />
      </picture>
    </div>
    <div
      className="emotion-20 emotion-21"
    >
      <div
        className="emotion-8 emotion-9"
      >
        MOW – HKT
      </div>
      <div
        className="emotion-10 emotion-11"
      >
        09:51 – 11:48
      </div>
      <div
        className="emotion-8 emotion-9"
      >
        В пути
      </div>
      <div
        className="emotion-10 emotion-11"
      >
        25ч 57м
      </div>
      <div
        className="emotion-8 emotion-9"
      >
        0 пересадок
      </div>
      <div
        className="emotion-10 emotion-11"
      >
        
      </div>
    </div>
    <div
      className="emotion-20 emotion-21"
    >
      <div
        className="emotion-8 emotion-9"
      >
        MOW – HKT
      </div>
      <div
        className="emotion-10 emotion-11"
      >
        14:39 – 03:11
      </div>
      <div
        className="emotion-8 emotion-9"
      >
        В пути
      </div>
      <div
        className="emotion-10 emotion-11"
      >
        12ч 32м
      </div>
      <div
        className="emotion-8 emotion-9"
      >
        3 пересадки
      </div>
      <div
        className="emotion-10 emotion-11"
      >
        BKK, AUH, HKG
      </div>
    </div>
  </div>
</div>
`;
